akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -2324,6 +2479,34 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
     }
   }
 
+  private def cleanupExpired(): Unit = {
+    if (expiryEnabled) {
+      val now = System.currentTimeMillis()
+      val expiredKeys = dataEntries.collect {
+        // it can be 0L when it was set via a DeltaPropagation or Write first time, don't expire such immediately
+        case (key, (_, _, usedTimestamp)) if usedTimestamp != 0L &&
+          getExpiryDuration(key) != Duration.Zero &&
+          usedTimestamp <= now - getExpiryDuration(key).toMillis =>
+          key
+      }
+
+      if (expiredKeys.nonEmpty) {
+        if (log.isDebugEnabled)
+          log.debug("Removing expired keys [{}]", expiredKeys.mkString(", "))
+
+        // Notification of subscribers for expired entries is not supported (yet)
```

Comment:
Seems useful to have/be able to observe from the outside.

Reply:
added notification of `Expired` in 8c27580d92

Reply:
Also remove the comment

```suggestion
```